packages/components/inputs/rich-text-utils/src/is-empty/is-empty.ts

```diff
@@ -1,6 +1,44 @@
+import {
+  Text,
+  Element,
+  type Text as TText,
+  type Element as TElement,
+} from 'slate';
 import html from '../html';
 
-const isEmpty = (rawValue: string): boolean =>
-  html.deserialize(rawValue).document.text === '';
+const isTextNodeNonEmpty = (node: TText) =>
+  Text.isText(node) && node.text !== '';
+
+const isElementNodeNonEmpty = (node: TElement) =>
+  node.children.some(isTextNodeNonEmpty);
```

Comment:
I'm a bit confused here since slate's Element.children prop seems to be defined as `Descendant[]` and that type is defined as `Element | Text` so we're not always passing `isTextNodeNonEmpty` function a `Text` typed object.

Reply:
@kark Do you still want to adjust this?

Reply:
I suppose that has already been covered but in 5e0bfd3b43d770c6f23921c12ef834915eb7868e I adjusted the use of `isElementNodeNonEmpty` within `isEmpty` so it is more straightforward